cpp/src/tests/connect/con_comp_test.cu

```diff
@@ -0,0 +1,180 @@
+    if (PERF) {
+     for (unsigned int i = 0; i < weakly_cc_time.size(); ++i) {
+      std::cout <<  weakly_cc_time[i] << std::endl;
+     }
+    } 
+  }
+  virtual void SetUp() {  }
+  virtual void TearDown() {  }
+
+  static std::vector<double> weakly_cc_time;
+
+  void run_current_test(const Usecase& param) {
+    const ::testing::TestInfo* const test_info =::testing::UnitTest::GetInstance()->current_test_info();
+    std::stringstream ss; 
+    std::string test_id = std::string(test_info->test_case_name()) + std::string(".") + std::string(test_info->name()) + std::string("_") + getFileName(param.get_matrix_file())+ std::string("_") + ss.str().c_str();
+    cudaStream_t stream{nullptr};
+
+#ifndef DEBUG_NO_RMM
+    rmmInitialize(nullptr);
+#endif
+
+    int m, k, nnz; //
+    MM_typecode mc;
+     
+    HighResClock hr_clock;
+    double time_tmp;
+
+    FILE* fpin = fopen(param.get_matrix_file().c_str(),"r");
+
+    ASSERT_EQ(mm_properties<int>(fpin, 1, &mc, &m, &k, &nnz),0) << "could not read Matrix Market file properties"<< "\n";
+    ASSERT_TRUE(mm_is_matrix(mc));
+    ASSERT_TRUE(mm_is_coordinate(mc));
+    ASSERT_TRUE(mm_is_symmetric(mc));//weakly cc only works w/ undirected graphs, for now;
+    
+    // Allocate memory on host
+    std::vector<int> cooRowInd(nnz);
+    std::vector<int> cooColInd(nnz);
+    std::vector<int> cooVal(nnz);
+    std::vector<int> labels(m);//for G(V, E), m := |V|
+
+    // Read
+    ASSERT_EQ( (mm_to_coo<int,int>(fpin, 1, nnz, &cooRowInd[0], &cooColInd[0], &cooVal[0], NULL)) , 0)<< "could not read matrix data"<< "\n";
+    ASSERT_EQ(fclose(fpin),0);
+
+    gdf_graph_ptr G{new gdf_graph, gdf_graph_deleter};
+    gdf_column_ptr col_src, col_dest, col_labels;
+
+    col_src = create_gdf_column(cooRowInd);
+    col_dest = create_gdf_column(cooColInd);
+    col_labels = create_gdf_column(labels);
+
+    ASSERT_EQ(gdf_adj_list_view(G.get(), col_src.get(), col_dest.get(), nullptr),0);
+
+    gdf_error status;
+    if (PERF)
+      {
+        hr_clock.start();
+        status = gdf_connected_components(G.get(),
+                                          col_labels.get(),
+                                          CUGRAPH_WEAK);
+
+        cudaDeviceSynchronize();
+        hr_clock.stop(&time_tmp);
+        weakly_cc_time.push_back(time_tmp);
+      }
+    else
+      {
+        cudaProfilerStart();
+        status = gdf_connected_components(G.get(),
+                                          col_labels.get(),
+                                          CUGRAPH_WEAK);
+        cudaProfilerStop();
+        cudaDeviceSynchronize();
+      }
+    EXPECT_EQ(status,GDF_SUCCESS);
+    
+#ifdef DEBUG_NO_RMM
+    rmmFinalize();
+#endif
+  }
+};
+ 
+std::vector<double> Tests_Weakly_CC::weakly_cc_time;
+
+TEST_P(Tests_Weakly_CC, Weakly_CC) {
+    run_current_test(GetParam());
+}
+
+// --gtest_filter=*simple_test*
+INSTANTIATE_TEST_CASE_P(simple_test, Tests_Weakly_CC, 
+			  ::testing::Values( Usecase("networks/dolphins.mtx")
```

Comment:
Should we add more test datasets? `dolphins.mtx` fits in a single thread block so it may miss some issues.  I would recommend other larger undirected datasets like :

```
test/datasets/coPapersDBLP.mtx
test/datasets/coPapersCiteseer.mtx
test/datasets/hollywood.mtx
```

Reply:
Are these undirected?

Reply:
yes

Reply:
There are a couple of options if you want to compare the results at the C++ level. Pagerank and SSSP load result files (generated by Nx and stored on AWS). SNMG SpMV compares against a trivial CPU implementation. This is a bit time consuming to generate and deploy. Since the code is already in production in cuML this is probably fine to skip it if you want to. 

A good tradeoff would be to just check that the output is not corrupted, we have something like this in another test:
```
        ASSERT_FALSE(isnan(calculated_res[i])); 
        ASSERT_LE(calculated_res[i], label_max);
        ASSERT_GE(calculated_res[i], label_min);
```

Reply:
Another thing we can do (and I've done that in the past w/ nvgraph) is to run networkx on same graph, _outside_ of our testing environment; dump results in a text file; then from C++ load the results file and use that as baseline. But that only works acceptably w/ small graphs.

Reply:
Yes, that's the first option I mentioned above, ie. what Pagerank and SSSP do. It is a bit long to deploy but that's the safest option.